### DISCUSSION TITLE
是否可以使用自己的用户系统

### DISCUSSION THREAD

**Discussion by AnzhiZhang** [walinejs/waline#1795] [Ideas]
集成到网站中时，希望可以使用自己的用户系统，避免用户在 waline 的用户系统中注册和登录

能否通过自定义 oauth 或其他方式实现？

相关issue：#908

---

有一种思路，不知道是否可行，麻烦帮忙提供一些建议
前端魔改禁用登录和退出按钮
后端鉴权：
1. 提供一个假数据库，由自己的用户系统同时维护，使用waline的算法
2. 如果服务端 Hooks 修改需要鉴权的操作，并且可以跳过自带的鉴权，似乎也是可行的
3. 自己实现一套兼容 waline api 的后端

**Comment by lizheming** [comment]
有这个需求的计划，不过这是一个比较长期的 todo 了，等我再想想吧

**Comment by lizheming** [comment]
https://waline.js.org/cookbook/customize/userdb.html 针对该需求出了一个版本，可以看下文档。没有实际测试过，有问题的话可以随时反馈。
